data-raw/covid_pmfs.R

```diff
@@ -0,0 +1,217 @@
+  # for the Delta variant and 4.2 d (95% CI: 3.6 to 4.9 d) for the Omicron
+  # variant Fig. 3B)."
+
+  # Fits a Weibull to the data
+
+  # Relies on fundamental assumption about epidemic growth rate.
+
+
+  discr_gr_adj_weibull <- tibble::tibble(
+    time = seq(0, 23, by = 1), # 23 seems to get most of the distribution mass
+    density0 = dweibull(time,
+      shape = backward_shape,
+      scale = backward_scale
+    ) * exp(r * time)
+  )
+
+  inc_period_pmf <- wwinference::to_simplex(discr_gr_adj_weibull$density0)
+  return(inc_period_pmf)
+}
+
+
+#' @title Make hospital onset delay pmf
+#' @description Uses the parameter estimates from cfa-parameter-estimates,
+#' which is based on Danache et al linelist data from symptom onset to hospital
+#' admission. See below:
+#' https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0261428
+#'
+#' @param neg_binom_mu float indicating the mean of the negative binomial shaped
+#' delay from symptom onset to hospital admissions, default is `6.98665` from
+#' fit to data in above paper
+#' @param neg_binom_size float indicating the dispersion parameter in the
+#' negative binomial delay from symptom onset to hospital admissions, default
+#' is `2.490848` from fit to data in above paper
+#'
+#' @return pmf of distribution from symptom onset to hospital admission
+#'
+#' @examples
+#' delay_pmf <- make_hospital_onset_delay_pmf(7, 2.5)
+make_hospital_onset_delay_pmf <- function(neg_binom_mu = 6.98665,
+                                          neg_binom_size = 2.490848) {
+  density <- dnbinom(
+    x = seq(0, 30, 1),
+    mu = neg_binom_mu, size = neg_binom_size
+  )
+  hosp_onset_delay_pmf <- density / sum(density)
+
+  return(hosp_onset_delay_pmf)
+}
+
+
+
+#' @title Make reporting delay pmf
+#' @description
+#' Convolve the incubation period pmf with the symptom to hospital admission pmf
+#' and normalize
+#'
+#' @param incubation_period_pmf a numeric vector, sums to 1, indicating
+#' the probability of time from infection to symptom onset
+#' @param hospital_onset_delay_pmf a numeric vector, sums to 1, indicating the
+#' proabbility of time from symptom onset to hospital admissions
+#'
+#' @return convolution of incubation period and sympton onset to hospital
+#' admission pmf
+#'
+#' @examples
+#' inc_pmf <- make_incubation_period_pmf(3.6, 1.5, 0.15)
+#' hosp_delay_pmf <- make_hospital_onset_delay_pmf(7, 2.5)
+#' inf_to_hosp_pmf <- make_reporting_delay_pmf(inc_pmf, hosp_delay_pmf)
+make_reporting_delay_pmf <- function(incubation_period_pmf,
+                                     hospital_onset_delay_pmf) {
+  pmfs <- list(
+    "incubation_period" = incubation_period_pmf,
+    "hosp_onset_delay" = hospital_onset_delay_pmf
+  )
+
+  infection_to_hosp_delay_pmf <- add_pmfs(pmfs) |>
+    wwinference::to_simplex()
+  return(infection_to_hosp_delay_pmf)
+}
+
+
+
+
+
+# Put it all together
+generation_interval <- withr::with_seed(42, {
+  wwinference::simulate_double_censored_pmf(
```

Comment:
This one didn't get updated
```suggestion
  simulate_double_censored_pmf(
```